### PR TITLE
Allow users to update schema without restarting the connection

### PR DESCRIPTION
**Pull request by rayman1104**
**What has been done? Why? What problem is being solved?**

If new space was created while the program is running, the schema is not updated and hence it's impossible to make queries to new spaces.

Right now if one wants to update the schema, they have to restart the connection completely, which is suboptimal. Making this `loadSchema` method public would allow them to trigger it manually at least. But of course in the future would be nice to update the schema automatically by the response from tarantool, as it's described in #7.

### CODE REVIEW

**Review by oleg-jukovec — CHANGES_REQUESTED** [review]
Thank you for the pull request! **We need tests for the pull request, since this functionality will be available in the public api.**

If you are blocked by the issue, you could manually update the schema right now with a workaround:

```Go
schema := new(tarantool.Schema)
schema.SpacesById = make(map[uint32]*tarantool.Space)
schema.Spaces = make(map[string]*tarantool.Space)

// Reload spaces.
var spaces []*tarantool.Space
err = conn.SelectTyped(vspaceSpId, 0, 0, maxSchemas, IterAll, []interface{}{}, &spaces)
if err != nil {
	return err
}
for _, space := range spaces {
	schema.SpacesById[space.Id] = space
	schema.Spaces[space.Name] = space
}

// Reload indexes.
var indexes []*Index
err = conn.SelectTyped(vindexSpId, 0, 0, maxSchemas, IterAll, []interface{}{}, &indexes)
if err != nil {
	return err
}
for _, index := range indexes {
	schema.SpacesById[index.SpaceId].IndexesById[index.Id] = index
	schema.SpacesById[index.SpaceId].Indexes[index.Name] = index
}
conn.OverrideSchema(schema) // or conn.Schema = schema
```

But be careful, this is not safe for active requests:

https://github.com/tarantool/go-tarantool/issues/218
https://github.com/tarantool/go-tarantool/pull/258/commits/35939df308be7fc7553080a823c5032536936138

Therefore, I will be able to merge this pull request only after #258 .

**Comment by kokizzu** [comment]
on code above what is `vspaceSpId`?

**Comment by oleg-jukovec** [comment]
> on code above what is `vspaceSpId`?

[281](https://www.tarantool.io/ru/doc/latest/reference/reference_lua/box_space/_vspace/)

vindexSpId:

[289](https://www.tarantool.io/ru/doc/latest/reference/reference_lua/box_space/_vindex/)

You could execute in Tarantool:

```
tarantool> box.space._space:select()
---
...
  - [281, 1, '_vspace', 'sysview', 0, {}, [{'name': 'id', 'type': 'unsigned'}, {'name': 'owner',
        'type': 'unsigned'}, {'name': 'name', 'type': 'string'}, {'name': 'engine',
        'type': 'string'}, {'name': 'field_count', 'type': 'unsigned'}, {'name': 'flags',
        'type': 'map'}, {'name': 'format', 'type': 'array'}]]
...
  - [289, 1, '_vindex', 'sysview', 0, {}, [{'name': 'id', 'type': 'unsigned'}, {'name': 'iid',
        'type': 'unsigned'}, {'name': 'name', 'type': 'string'}, {'name': 'type',
        'type': 'string'}, {'name': 'opts', 'type': 'map'}, {'name': 'parts', 'type': 'array'}]]
...
```

**Comment by rayman1104** [comment]
@oleg-jukovec can we merge it now since https://github.com/tarantool/go-tarantool/pull/258 is merged?

**Comment by oleg-jukovec** [comment]
> @oleg-jukovec can we merge it now since #258 is merged?

First you need to add tests for the public API. At least you need to add an integration test with Tarantool to `tarantool_test.go`.

**Comment by oleg-jukovec** [comment]
The pull request closed due to inactivity. You can re-open it with tests for the public API.

**Pull request closed without merging**
